corehq/messaging/smsbackends/amazon_pinpoint/views.py

```diff
@@ -0,0 +1,35 @@
+import json
+from urllib.request import urlopen
+from corehq.apps.sms.api import incoming as incoming_sms
+from corehq.apps.sms.views import IncomingBackendView
+from corehq.messaging.smsbackends.amazon_pinpoint.models import SQLPinpointBackend
+from django.http import HttpResponse
+
+
+class PinpointIncomingMessageView(IncomingBackendView):
+    urlname = 'pinpoint_sms'
+
+    @property
+    def backend_class(self):
+        return SQLPinpointBackend
+
+    def post(self, request, api_key, *args, **kwargs):
+        request_body = json.loads(request.body)
+        message = request_body.get('Message')
+        subscribe_url = request_body.get('SubscribeURL')
+        if subscribe_url:
+            urlopen(subscribe_url)
```

Comment:
What does this do?

Reply:
As soon as webhook is attached, Amazon SNS sends a subscription confirmation message to the endpoint. The code at the endpoint must retrieve the SubscribeURL from the subscription confirmation message and then either visit the location specified by the SubscribeURL or make it available to developer (or admin user) so that he can manually visit the SubscribeURL, using a web browser.